src/helpers/throttle.ts

```diff
@@ -0,0 +1,24 @@
+export default function throttle<A = unknown, R = void>(
+  fn: (args: A) => R,
+  ms: number
+): [(args: A) => Promise<R>, () => void] {
+  let isThrottled = false;
+
+  const throttledFunc = (args: A): Promise<R> =>
+    new Promise((resolve) => {
+      if (!isThrottled) {
+        isThrottled = true;
+        resolve(fn(args));
+
+        setTimeout(() => {
+          isThrottled = false;
+        }, ms);
```

Comment:
Hi @wxharry, I think this implementation of throttle has a bug: if the `throttledFunc` is continuously called every `ms-0.000...1` ms, then the `fn` will never be called. I mean, once the `isThrottled` is set to `true`, succeeding calls of `throttledFunc` should not refresh the `isThrottled`.

Don't worry, I suppose these basic functions may not be written by our own. I'll push a commit to substitue them with [lodash's](https://lodash.com/docs/4.17.15#throttle).